routes/auth.js

```
@@ -0,0 +1,69 @@
+const express = require('express')
+const bcrypt = require('bcryptjs')
+const {check, validationResult} = require('express-validator')
+const jwt = require('jsonwebtoken')
+const config = require('config')
+const auth = require('../middleware/auth')
+const User = require('../models/User')
+const router = express.Router()
+
+// Get user
+router.get('/', auth, async (req, res) => {
+  try {
+    const user = await User.findById(req.user.id).select('-password')
```

In general, I prefer whitelisting instead of blacklisting. In other words, instead of saying, "everything except password" I prefer "give me email and name".

This is because say in the future a developer adds to the schema an attribute to user called, "social security number". They might know that this part of the code returns everything and we'll accidentally expos the social security number.

---

Ok.